accountsdb-plugin-manager/src/accountsdb_plugin_service.rs

```diff
@@ -145,12 +159,14 @@ impl AccountsDbPluginService {
         Ok(())
     }
 
-    pub fn get_accounts_update_notifier(&self) -> AccountsUpdateNotifier {
+    pub fn get_accounts_update_notifier(&self) -> Option<AccountsUpdateNotifier> {
         self.accounts_update_notifier.clone()
     }
 
-    pub fn join(mut self) -> thread::Result<()> {
-        self.slot_status_observer.join()?;
+    pub fn join(self) -> thread::Result<()> {
+        if self.slot_status_observer.is_some() {
+            self.slot_status_observer.unwrap().join()?;
+        }
```

Comment:
```suggestion
        if let Some(slot_status_observer) = self.slot_status_observer {
            slot_status_observer.join()?;
        }
```
This would be a rust-ier way to do this. I'm surprised clippy didn't suggest it, actually...

Reply:
done